modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/mapping/MappingServiceImpl.java

```diff
@@ -409,4 +413,31 @@ public int hashCode() {
             return Objects.hash(planId, mapOnBackups);
         }
     }
+
+    private static class DistributionHolder {
+        private final List<String> nodes;
+        private final Map<Integer, List<TokenizedAssignments>> assignmentsPerTable;
+        private final Map<String, List<String>> nodesPerView;
```

Comment:
```suggestion
        private final Int2ObjectMap<List<String>> nodesPerView;
```
This also can be a int->obj map.